src/amp-story-player/amp-story-player-impl.js

```diff
@@ -905,11 +929,16 @@ export class AmpStoryPlayer {
   }
 
   /**
-   * Reacts to touchstart events and caches its coordinates.
+   * Reacts to touchstart events, caches its coordinates,
+   * and ends the autoplay mode if activated.
    * @param {!Event} event
    * @private
    */
   onTouchStart_(event) {
+    if (this.isAutoplaying_) {
```

Comment:
Doing this check in `touchStart_` won't work for desktop.

Reply:
Where could it go instead?

Reply:
After looking further at the code, there doesn't seem to be any way to detect any type of user interaction. `onTouchStart_` only works on touchscreen, and `onSelectDocument_` fires any time the story is changed (regardless of whether there has been any interaction)

Maybe I could set up a one-time event listener for click/touch on the element handling this?

Reply:
Yep, I think we need to set up a click listener at the player level, so that we can catch this kind of interactions.

Reply:
I tried a new approach on the latest commits. How is it looking?